pkg/client/unversioned/request.go

```diff
@@ -683,9 +689,13 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 		return nil, r.err
 	}
 
+	now := time.Now()
 	if r.throttle != nil {
 		r.throttle.Accept()
 	}
+	if latency := time.Since(now); latency > longThrottleLatency {
+		glog.Warningf("Throttling request took %v, request: %s", latency, r.URL().String())
+	}
```

Comment:
Perhaps replace these 7 lines with a single function, r.tryThrottle or some such, to reduce code duplication? Also you could put the timing code in side the if statement, but that's a silly nit.